### PR TITLE
Post Video Recording Edits.

### PR DESCRIPTION
**Pull request by ashok2811**
- Fast and Efficient ways to capture and calibrate single and stereo cameras
- Stereo validation Updated

### CODE REVIEW

**Comment by ashok2811** [comment]
- Please Update the `milestones` of the project.

**Comment by surajhanchinal** [comment]
Remove changes in the C++ source files. Clean up the python scripts, remove unused config files

File: scripts/single_camera_calibrate.py [hunk 1]
```diff
@@ -64,15 +74,25 @@ def calibrate_camera(images_folder, camera_name):
         print("ROI: ", roi)
 
         # Save the calibration parameters to a file
-        np.savez(camera_name + "_intrinsic", mtx=mtx, dist=dist, newCameraMatrix=newCameraMatrix)
+        #np.savez(camera_name + "_intrinsic", mtx=mtx, dist=dist, newCameraMatrix=newCameraMatrix)
```
Comment (surajhanchinal) [inline scripts/single_camera_calibrate.py]: remove this

File: scripts/stereoMap.xml
Comment (surajhanchinal) [inline scripts/stereoMap.xml:1]: delete this file

File: scripts/stereo_camera_calibration.py [hunk 2]
```diff
@@ -11,16 +12,43 @@
...
+    else :
+        x = math.atan2(-R[1,2], R[1,1])
+        y = math.atan2(-R[2,0], sy)
+        z = 0
+ 
+    return np.array([x*(180/math.pi), y*(180/math.pi), z*(180/math.pi)])
+
+scale_factor = 0.7
+
 imgs = [6,7,21,22,28,29,30,33,46,50,51,52,53,54,55,56,57,58,59,60]
```
Comment (surajhanchinal) [inline scripts/stereo_camera_calibration.py]: delete

File: scripts/stereo_camera_calibration.py [hunk 3]
```diff
@@ -11,16 +12,43 @@
...
 
 objPoints = []
 imgPointsLeft = []
 imgPointsRight = []
 
+ignore_list = [18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,84]
+
 while True:
+    print(counter2)
```
Comment (surajhanchinal) [inline scripts/stereo_camera_calibration.py]: delete

File: scripts/stereo_camera_calibration.py [hunk 4]
```diff
@@ -11,16 +12,43 @@
...
     leftName = './images/calibration/left/' + str(counter2) + '.jpg'
     rightName = './images/calibration/right/' + str(counter2) + '.jpg'
-    counter2= counter2 + 1
+    skip = False
+    for x in ignore_list:
+        if x == counter2:
+            skip = True
+            break
+    if skip:
+        print("skipping: ",counter2)
```
Comment (surajhanchinal) [inline scripts/stereo_camera_calibration.py]: delete

File: scripts/stereo_camera_calibration.py [hunk 5]
```diff
@@ -32,64 +60,83 @@
...
 
+flags |= cv2.CALIB_USE_EXTRINSIC_GUESS
 
-criteria_stereo= (cv2.TERM_CRITERIA_EPS + cv2.TERM_CRITERIA_MAX_ITER, 30, 0.001)
+criteria_stereo= (cv2.TERM_CRITERIA_EPS + cv2.TERM_CRITERIA_MAX_ITER, 1000, 0.001)
 
-retStereo, newCameraMatrixL, distL, newCameraMatrixR, distR, rot, trans, essentialMatrix, fundamentalMatrix = cv2.stereoCalibrate(objPoints, imgPointsLeft, imgPointsRight, newCameraMatrixL, distL, newCameraMatrixR, distR, grayLeft.shape[::-1], criteria_stereo, flags)
+initR = np.identity(3).astype(np.float64)
+initT = np.array([[820],[0],[0]],dtype=np.float64)
+print(initR.shape)
```
Comment (surajhanchinal) [inline scripts/stereo_camera_calibration.py]: delete print here and below

File: scripts/stereo_camera_calibration.py [hunk 6]
```diff
@@ -32,64 +60,83 @@
...
+print(initT.shape)
+print(grayLeft.shape[::-1])
 
+#retStereo, newCameraMatrixL, distL, newCameraMatrixR, distR, rot, trans, essentialMatrix, fundamentalMatrix = cv2.stereoCalibrate(objPoints, imgPointsLeft, imgPointsRight, newCameraMatrixL, distL, newCameraMatrixR, distR, grayLeft.shape[::-1], criteria_stereo, flags)
+ttp = cv2.stereoCalibrateExtended(objPoints, imgPointsLeft, imgPointsRight,newCameraMatrixL,distL,newCameraMatrixR,distR,grayLeft.shape[::-1],initR,initT,flags=flags)
+print(len(ttp))
```
Comment (surajhanchinal) [inline scripts/stereo_camera_calibration.py]: delete

File: scripts/single_camera_calibrate.py (hunk 1 above)
Comment (ashok2811) [reply scripts/single_camera_calibrate.py]: Done!

File: scripts/stereo_camera_calibration.py (hunk 4 above)
Comment (ashok2811) [reply scripts/stereo_camera_calibration.py]: Done!

File: scripts/stereo_camera_calibration.py (hunk 5 above)
Comment (ashok2811) [reply scripts/stereo_camera_calibration.py]: Done!

File: scripts/stereo_camera_calibration.py (hunk 2 above)
Comment (ashok2811) [reply scripts/stereo_camera_calibration.py]: Done!

File: scripts/stereo_camera_calibration.py (hunk 3 above)
Comment (ashok2811) [reply scripts/stereo_camera_calibration.py]: Done

File: scripts/stereo_camera_calibration.py (hunk 6 above)
Comment (ashok2811) [reply scripts/stereo_camera_calibration.py]: Done!

**Pull request merged**